lifecycle/README.rst

```diff
@@ -0,0 +1,343 @@
+   [lc_client] Node lc_talker has current state active.
+
+makes the lifecycle talker change its state to active.
+Active means that all publishers and timers are now activated.
+Therefore the messages are now getting published.
+
+.. code-block:: bash
+
+   [lc_talker] on_activate() is called.
+   [lc_talker] Lifecycle publisher is active. Publishing: [Lifecycle HelloWorld #11]
+   [lc_talker] Lifecycle publisher is active. Publishing: [Lifecycle HelloWorld #12]
+   ...
+
+The lifecycle listener receives the same set of notifications as before.
+Lifecycle talker changed its state from inactive to active.
+
+.. code-block:: bash
+
+   [lc_listener] notify callback: Transition from state unconfigured to configuring
+   [lc_listener] notify callback: Transition from state configuring to inactive
+
+The difference to the transition event before is that our listener now also receives the actual published data.
+
+.. code-block:: bash
+
+   [lc_listener] data_callback: Lifecycle HelloWorld #11
+   [lc_listener] data_callback: Lifecycle HelloWorld #12
+   ...
+
+Please note that the index of the published message is already at 11.
+The purpose of this demo is to show that even though we call ``publish`` at every state of the lifecycle talker, only when the state in active, the messages are actually published.
+As for the beta1, all other messages are getting ignored.
+This behavior may change in future versions in order to provide better error handling.
+
+For the rest of the demo, you will see similar output as we deactivate and activate the lifecycle talker and finally shut it down.
+
+The demo code
+-------------
+
+lifecycle_talker, lifecycle_listener and lifecycle_service_client
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If we have a look at the code, there is one significant change for the lifecycle talker compared to a regular talker.
+Our node does not inherit from the regular ``rclcpp::node::Node`` but from ``rclcpp_lifecycle::LifecycleNode``.
+
+.. code-block:: bash
+
+   class LifecycleTalker : public rclcpp_lifecycle::LifecycleNode
+
+Every child of LifecycleNodes have a set of callbacks provided.
+These callbacks go along with the applied state machine attached to it.
+These callbacks are:
+
+
+* ``rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+    on_configure(const rclcpp_lifecycle::State & previous_state)``
+* ``rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+    on_activate(const rclcpp_lifecycle::State & previous_state)``
+* ``rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+    on_deactivate(const rclcpp_lifecycle::State & previous_state)``
+* ``rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+    on_cleanup(const rclcpp_lifecycle::State & previous_state)``
+* ``rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn
+    on_shutdown(const rclcpp_lifecycle::State & previous_state)``
```

Comment:
The functions look a bit weird on github's preview, but the bouncy version looks fine https://github.com/ros2/demos/blob/bouncy/lifecycle/README.rst

Reply:
should be addressed now